Lib/python/builtin.swg

```diff
@@ -403,16 +403,14 @@ SwigPyBuiltin_InitBases (PyTypeObject *type, PyTypeObject **bases) {
     bases[0] = SwigPyObject_type();
     bases[1] = NULL;
   }
-  type->tp_base = bases[0];
-  Py_INCREF((PyObject *)bases[0]);
   for (b = bases; *b != NULL; ++b)
     ++base_count;
   tuple = PyTuple_New(base_count);
```

Comment:
Like almost any Python API, `PyTuple_New` can fail. But it seems SWIG generally doesn't handle errors, is that right?

Reply:
The generated wrappers ideally should handle errors, but there are certainly cases where currently they don't.  A particular case that comes to mind is there's not much checking for `malloc()` failing, which really isn't great but it's also a situation where it's often hard to gracefully recover - recovery likely means further calls to the target language API which generally will end up trying to allocate memory, which is likely to also fail unless the original failed allocation was large (and calls to `malloc()` in the generated wrapper code will generally be for structures, etc and so small).  E.g. see #277.

It looks like there's already missing checking here so good to highlight but could reasonably be addressed separately.